tools/pycafe/create_pycafe_links.py

```diff
@@ -0,0 +1,131 @@
+    requirements = (
+        f"""{PYCAFE_URL}/gh/artifact/mckinsey/vizro/actions/runs/{RUN_ID}/pip/vizro-{PACKAGE_VERSION}-py3-none-any.whl\n"""
+        + extra_requirements_concat
+    )
+
+    # App file
+    app_file_path = os.path.join(directory, "app.py")
+    app_content = Path(app_file_path).read_text()
+    app_content_split = app_content.split('if __name__ == "__main__":')
+    if len(app_content_split) > 1:
+        app_content = app_content_split[0] + textwrap.dedent(app_content_split[1])
+
+    # JSON object
+    json_object = {
+        "code": str(app_content),
+        "requirements": requirements,
+        "files": [],
+    }
+    for root, _, files in os.walk("./" + directory):
+        for file in files:
+            if "app.py" in file:
+                continue
+            file_path = os.path.join(root, file)
+            relative_path = os.path.relpath(file_path, directory)
+            file_url = f"{base_url}{relative_path.replace(os.sep, '/')}"
+            json_object["files"].append({"name": relative_path, "url": file_url})
+
+    json_text = json.dumps(json_object)
+    compressed_json_text = gzip.compress(json_text.encode("utf8"))
+    base64_text = base64.b64encode(compressed_json_text).decode("utf8")
+    query = urlencode({"c": base64_text}, quote_via=quote)
+    return f"{PYCAFE_URL}/snippet/vizro/v1?{query}"
+
+
+def post_comment(urls: list[tuple[str, str]]):
+    """Post a comment on the pull request with the links to the PyCafe dashboards."""
+    # Inspired by https://github.com/snehilvj/dash-mantine-components
+
+    # Find existing comments by the bot
+    comments = pr.get_issue_comments()
+    bot_comment = None
+    for comment in comments:
+        if comment.body.startswith("View the example dashboards of the current commit live"):
+            bot_comment = comment
+            break
+
+    # Get current UTC datetime
+    current_utc_time = datetime.datetime.now(datetime.UTC).strftime("%Y-%m-%d %H:%M:%S UTC")
+
+    # Define the comment body with datetime
+    dashboards = "\n\n".join(f"Link: [{directory}]({url})" for url, directory in urls)
+
+    comment_body = f"""View the example dashboards of the current commit live on PyCafe:\n
+Updated on: {current_utc_time}
+Commit: {commit_sha}
+
+{dashboards}
+"""
+
+    # Update the existing comment or create a new one
+    if bot_comment:
+        bot_comment.edit(comment_body)
+        print("Comment updated on the pull request.")  # noqa
+    else:
+        pr.create_issue_comment(comment_body)
+        print("Comment added to the pull request.")  # noqa
+
+
+if __name__ == "__main__":
+    urls = []
+
+    # Generate links for each directory and create status
+    for directory in sys.argv[1:]:
+        # Add any extra requirements for specific dev examples
+        if directory == "examples/dev/":
+            url = generate_link(directory=directory, extra_requirements=["openpyxl"])
+        else:
+            url = generate_link(directory=directory)
```

Comment:
Why do we not need the extra requirements for the visual-vocabulary dashboard?

Reply:
But visual vocabulary depends on black and maybe some other things?

Reply:
Hm I check HF, and indeed we have black. Strange, not sure why...